documentation/src/main/asciidoc/user_guide/execute_grid.adoc

```diff
@@ -0,0 +1,189 @@
+==  Executing code in the Grid
+
+The main benefit of a Cache is the ability to very quickly lookup a value
+by its key, even across machines. In fact this use alone is probably the reason
+many users use Infinispan. However Infinispan can provide many more benefits
+that aren't immediately apparent. Since Infinispan is usually used in
+a cluster of machines we also have features available that can help utilize
+the entire cluster for performing the user's desired workload.
+
+NOTE: This section covers only executing code in the grid using an embedded cache,
+if you are using a remote cache you should check out
+<<executing_code_in_the_remote_grid, Executing code in the Remote Grid>>.
+
+=== Cluster Executor
+
+Since you have a group of machines, why not off load execution of code to them.
```

Comment:
"off-load".
Also I don't like the faux-question style, so I'd turn the sentence to something like "Since you have a group of machines, it makes sense to leverage their combined computing power for executing code on all of them".